clients/Android/ultrablue/app/src/main/java/fr/gouv/ssi/ultrablue/model/UltrablueProtocol.kt

```diff
@@ -160,36 +165,45 @@ class UltrablueProtocol(private val activity: MainActivity, private var enroll:
                     logger?.push(CLog("Error while getting PCRs: ${e.message}", false))
                     return
                 }
-                val response: AttestationResult = if (enroll) {
+                attestationResponse = if (enroll) {
                     logger?.push(Log("Storing new attester entry"))
                     registerDevice()
-                    AttestationResult(false, "")
+                    AttestationResponse(false, device.secret)
                 } else {
                     logger?.push(Log("Comparing PCRs"))
                     if (device.encodedPCRs.contentEquals(encodedPCRs)) {
                         logger?.push(CLog("PCRs entries match the stored ones", true))
-                        AttestationResult(false, "")
+                        AttestationResponse(false, device.secret)
                     } else {
                         // TODO: Need deeper investigation to determine which PCR changed and why.
                         logger?.push(CLog("PCRs don't match", false))
-                        AttestationResult(true, "Attestation failure")
+                        AttestationResponse(true, byteArrayOf())
                     }
                 }
-                val encodedResponse = Cbor.encodeToByteArray(response)
-                writeMsg("Attestation result", encodedResponse)
+                state = RESPONSE
+                resume()
             }
-            END -> {
-                return
+            RESPONSE -> {
+                val encodedResponse = Cbor.encodeToByteArray(attestationResponse)
+                writeMsg("Attestation response", encodedResponse)
             }
         }
     }
 
     private fun registerDevice() {
+        val secret = if (ek.PCRExtend) {
+            ByteArray(16)
+        } else {
+            byteArrayOf()
+        }
+        rand.nextBytes(secret)
```

Comment:
It works but I find confusing the fact that you essentially "push zero random bytes" in the non-extend case. I'd find the following clearer:
val secret = byteArrayOf()
if (extend) { secret = ByteArray(16). rand.nextBytes(secret) }